### PR TITLE
Move probe host topology code into its own module

### PR DESCRIPTION
**Pull request by tomwilkie**
- add tests
- use constants for NodeMetadata keys
- add host uptime and kernel version

helps toward #238

### CODE REVIEW

File: probe/host/reporter.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	var result time.Duration
+
+	buf, err := ReadFile(ProcUptime)
+	if err != nil {
+		return result, err
+	}
+
+	fields := strings.Fields(string(buf))
+	if len(fields) != 2 {
+		return result, err
```
Comment (paulbellamy) [inline probe/host/reporter.go]: Did you mean to make a new error here?

File: probe/host/reporter.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+func (r *reporter) Report() (report.Report, error) {
+	var (
+		rep        = report.MakeReport()
+		localCIDRs []string
+		utsname    syscall.Utsname
+	)
+
+	if localNets, err := InterfaceAddrs(); err == nil {
```
Comment (paulbellamy) [inline probe/host/reporter.go]: Ignoring errors here?

**Comment by tomwilkie** [comment]
@paulbellamy PTAL

**Comment by paulbellamy** [comment]
lgtm

**Pull request merged**